connectors/sources/mssql.py

```diff
@@ -62,6 +62,244 @@ def all_schemas(self):
+
+    def create_pem_file(self):
+        """Create pem file for SSL Verification"""
+        pem_certificates = get_pem_format(key=self.ssl_ca)
+        with NamedTemporaryFile(mode="w", suffix=".pem", delete=False) as cert:
+            cert.write(pem_certificates)
+            self.certfile = cert.name
+
+    async def get_cursor(self, query):
+        """Executes the passed query on the Non-Async supported Database server and return cursor.
+
+        Args:
+            query (str): Database query to be executed.
+
+        Returns:
+            cursor: Synchronous cursor
+        """
+        try:
+            loop = asyncio.get_running_loop()
+            if self.connection is None:
+                self.connection = await loop.run_in_executor(
+                    executor=None, func=self.engine.connect  # pyright: ignore
+                )
+            cursor = await loop.run_in_executor(
+                executor=None,
+                func=partial(self.connection.execute, statement=text(query)),
+            )
+            return cursor
+        except Exception as exception:
+            self._logger.warning(
+                f"Something went wrong while executing query. Exception: {exception}"
+            )
+            raise
+
+    async def ping(self):
+        return await anext(
+            self._execute_query(
+                query=self.queries.ping(),
+            )
+        )
+
+    async def get_tables_to_fetch(self):
+        tables = configured_tables(self.tables)
+        return list(
+            map(
+                lambda table: table[0],  # type: ignore
+                await anext(
+                    self._execute_query(
+                        query=self.queries.all_tables(
+                            database=self.database,
+                            schema=self.schema,
+                        )
+                    )
+                ),
+            )
+            if is_wildcard(tables)
+            else tables
+        )
+
+    async def get_table_row_count(self, table):
+        [[row_count]] = await anext(
+            self._execute_query(
+                query=self.queries.table_data_count(
+                    schema=self.schema,
+                    table=table,
+                ),
+            )
+        )
+        return row_count
+
+    async def get_table_primary_key(self, table):
+        return await anext(
+            self._execute_query(
+                query=self.queries.table_primary_key(
+                    schema=self.schema,
+                    table=table,
+                ),
+            )
+        )
+
+    async def get_table_last_update_time(self, table):
+        return await anext(
+            self._execute_query(
+                query=self.queries.table_last_update_time(
+                    schema=self.schema,
+                    table=table,
+                ),
+            )
+        )
+
+    async def data_streamer(self, table):
```

Comment:
Maybe rename to `yield_table_data`?

Reply:
This name is actually from the original code:
https://github.com/elastic/connectors-python/blob/e9b4e577b4311d8fd8ca01cddae14ec5528ae9c7/connectors/sources/generic_database.py#L396-L404

WDYT about `table_data_generator`?

Reply:
I see, might be worth to keep it the way it was before too, not a biggie